src/ext-test/java/org/opentripplanner/ext/realtimeresolver/RealtimeResolverTest.java

```diff
@@ -0,0 +1,481 @@
+package org.opentripplanner.ext.realtimeresolver;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import gnu.trove.set.TIntSet;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.LineString;
+import org.opentripplanner.ext.flex.FlexIndex;
+import org.opentripplanner.model.FeedInfo;
+import org.opentripplanner.model.PathTransfer;
+import org.opentripplanner.model.StopTimesInPattern;
+import org.opentripplanner.model.Timetable;
+import org.opentripplanner.model.TripTimeOnDate;
+import org.opentripplanner.model.calendar.CalendarService;
+import org.opentripplanner.model.plan.Itinerary;
+import org.opentripplanner.model.plan.Leg;
+import org.opentripplanner.model.plan.Place;
+import org.opentripplanner.model.plan.legreference.LegReference;
+import org.opentripplanner.model.transfer.TransferService;
+import org.opentripplanner.routing.algorithm.raptoradapter.transit.TransitLayer;
+import org.opentripplanner.routing.services.TransitAlertService;
+import org.opentripplanner.routing.stoptimes.ArrivalDeparture;
+import org.opentripplanner.transit.model.basic.Notice;
+import org.opentripplanner.transit.model.basic.TransitMode;
+import org.opentripplanner.transit.model.framework.AbstractTransitEntity;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.network.GroupOfRoutes;
+import org.opentripplanner.transit.model.network.Route;
+import org.opentripplanner.transit.model.network.TripPattern;
+import org.opentripplanner.transit.model.organization.Agency;
+import org.opentripplanner.transit.model.organization.Operator;
+import org.opentripplanner.transit.model.site.AreaStop;
+import org.opentripplanner.transit.model.site.MultiModalStation;
+import org.opentripplanner.transit.model.site.RegularStop;
+import org.opentripplanner.transit.model.site.Station;
+import org.opentripplanner.transit.model.site.StopLocation;
+import org.opentripplanner.transit.model.site.StopLocationsGroup;
+import org.opentripplanner.transit.model.timetable.Trip;
+import org.opentripplanner.transit.model.timetable.TripIdAndServiceDate;
+import org.opentripplanner.transit.model.timetable.TripOnServiceDate;
+import org.opentripplanner.transit.service.TransitService;
+import org.opentripplanner.updater.GraphUpdaterStatus;
+
+class RealtimeResolverTest {
+
+  private static MockLeg MOCK_LEG = new MockLeg();
+
+  private static class MockLeg implements Leg {
```

Comment:
You can use leg implementations from `TestItineraryBuilder`

Reply:
Unfortunately not. I need custom implementation on `getLegReference`.

Reply:
Done